src/rule.hpp

```diff
@@ -16,36 +15,48 @@
 #include "event.hpp"
 #include "exclusion/common.hpp"
 #include "expression.hpp"
-#include "iterator.hpp"
 #include "matcher/base.hpp"
+#include "module_category.hpp"
 #include "object_store.hpp"
 
 namespace ddwaf {
 
-class rule {
+// A core rule constitutes the most important type of entity within the
+// evaluation process. These rules are "request-bound", i.e. they are used to
+// specifically analyse request data, as opposed to other types of rules such
+// as threshold rules which analyse data across requests.
```

Comment:
Do these threshold rules actually exist?

Reply:
Yes, in a different PR... I can dream...